src/dds/rs-dds-option.cpp

```diff
@@ -19,15 +22,38 @@ static option_range range_from_realdds( std::shared_ptr< realdds::dds_option > c
                  dds_opt->get_stepping(),
                  dds_opt->get_default_value() };
     }
+    if( std::dynamic_pointer_cast< realdds::dds_boolean_option >( dds_opt ) )
+    {
+        return { 0, 1, 1, bool( dds_opt->get_default_value() ) ? 1.f : 0.f };
+    }
+    if( auto e = std::dynamic_pointer_cast< realdds::dds_enum_option >( dds_opt ) )
+    {
+        return { 0, float( e->get_choices().size() - 1 ), 1, (float)e->get_value_index( e->get_default_value() ) };
+    }
     return { 0, 0, 0, 0 };
 }
 
 
+static rs2_option_type rs_type_from_dds_option( std::shared_ptr< realdds::dds_option > const & dds_opt )
+{
+    if( std::dynamic_pointer_cast< realdds::dds_float_option >( dds_opt ) )
+        return RS2_OPTION_TYPE_FLOAT;
+    if( std::dynamic_pointer_cast< realdds::dds_string_option >( dds_opt ) )
+        return RS2_OPTION_TYPE_STRING;
+    if( std::dynamic_pointer_cast< realdds::dds_boolean_option >( dds_opt ) )
+        return RS2_OPTION_TYPE_BOOLEAN;
+    if( std::dynamic_pointer_cast< realdds::dds_integer_option >( dds_opt ) )
+        return RS2_OPTION_TYPE_INTEGER;
+    return RS2_OPTION_TYPE_COUNT;
```

Comment:
What does returning count means? Won't it be better to throw "unsupported" or maybe return the default float type?

Reply:
It's just the type that would be returned to someone calling get_value_type() -- if you prefer, I can throw. Done.